wrappers/android/tools/camera/src/main/java/com/intel/realsense/camera/SettingsActivity.java

```diff
@@ -130,9 +130,10 @@ private void loadSettingsList(final Device device){
         settingsMap.put(INDEX_ADVANCE_MODE,"Enable advanced mode");
         if(device.is(Extension.UPDATABLE)){
             settingsMap.put(INDEX_UPDATE,"Firmware update");
-            Updatable fwud = device.as(Extension.UPDATABLE);
-            if(fwud != null && fwud.supportsInfo(CameraInfo.CAMERA_LOCKED) && fwud.getInfo(CameraInfo.CAMERA_LOCKED).equals("NO"))
-                settingsMap.put(INDEX_UPDATE_UNSIGNED,"Firmware update (unsigned)");
+            try(Updatable fwud = device.as(Extension.UPDATABLE)){
```

Comment:
Any other device.as() calls that you call quickly find?

Reply:
found 2 others!